### DISCUSSION TITLE
File uploads

### DISCUSSION THREAD

**Discussion by JohnHardy** [collerek/ormar#516] [Q&A]
Hello!  Loving Ormar!

I am migrating some ideas from a Django project and wondering the best way to manage file uploads (akin to `FileField`).  Have I missed some obvious documentation?

Is there a best practice for how to manage saving files using the `File` and `UploadFile` classes?  Perhaps similar to this: https://github.com/tortoise/tortoise-orm/issues/100#issuecomment-460038700

I had a quick look at how to add something myself to Ormar, but thought I would ask before starting, as I doubt I am the first to wonder. :)

Thanks!
John

**Comment by collerek** [comment]
If you want to keep the files in db (which usually is not a good practice) you can simply use binary column.

If you want to upload a file through API (assuming fastapi) you need to send form/multipart anyway like explained in example here: https://github.com/tiangolo/fastapi/issues/657

**Reply by JohnHardy** [reply]
Thanks for your help.  I'm not trying to store a file in the DB :) I was wondering how to manage the files being uploaded and saved to disk - it is something I need to do by hand?
